homeassistant/components/media_player/yamaha.py

```diff
@@ -110,10 +149,16 @@ def __init__(self, name, receiver, source_ignore, source_names):
         self._name = name
         self._zone = receiver.zone
 
+    def __eq__(self, other):
```

Comment:
Instead of adding this method, implement the property `unique_id` and let the Entity base class do the work for you 👍